system-addon/content-src/components/Search/Search.jsx

```diff
@@ -0,0 +1,34 @@
+"use strict";
+const React = require("react");
+const {connect} = require("react-redux");
+const {actionTypes, actionCreators} = require("common/Actions.jsm");
+
+const Search = React.createClass({
+  getInitialState() {
+    return {
+      searchString: ''
+    };
+  },
+  performSearch(options) {
+    let searchData = {
+      engineName: options.engineName,
+      searchString: options.searchString,
+      searchPurpose: "d",
+      healthReportKey: "newtab"
+    };
+    this.props.dispatch(actionCreators.SendToMain({type: actionTypes.PERFORM_SEARCH, data: searchData}));
+  },
+  render() {
+    const {currentEngine, engines} = this.props.Search;
+    return (<form className="search-wrapper">
+      <span className="search-label" />
+      <input value={this.state.searchString} type="search"
+        onChange={e => this.setState({searchString: e.target.value})}
+        maxLength="256"
+        placeholder="Search the Web"/>
```

Comment:
I see the plain string in `TopSites`, so do we have a good place to track these strings to replace with #2313?